packages/assets-controllers/src/NftController.ts

```diff
@@ -98,6 +98,10 @@ type NftUpdate = {
   newMetadata: NftMetadata;
 };
 
+export type NftFetchStatus = {
+  isFetchingInProgress: boolean;
```

Comment:
Is this for:
1. extension/mobile clients to query?
2. prevent concurrent updates?

If (1), can clients already know this by waiting for `detectNfts` to return, since they're now responsible for calling it?

if (2), a mutex instead of state?

Reply:
I think there's an alternative we could consider now or in a future refactor, achieving the same but in a stateless way.

In the case of a single visit, I think the client would know it's done because it triggered the fetch:

```ts
const [loading, setLoading] = useState(true);
useEffect(() => {
    await nftDetectionController.detectNfts(...);
    setLoading(false);
}, [])
```

But if you switch back and forth between the NFT tab, it can't distinguish the possibility that it hit [this no-op](https://github.com/MetaMask/core/blob/a0687fe4e03ef1e47a13604757297760cb0dd117/packages/assets-controllers/src/NftDetectionController.ts#L520) because a previous fetch is in progress.  It needs a way to hook into the fetch triggered by a previous render.

It could do what the token rates controller does.  Store a private variable (no state) of the [ongoing promises](https://github.com/MetaMask/core/blob/a0687fe4e03ef1e47a13604757297760cb0dd117/packages/assets-controllers/src/TokenRatesController.ts#L169), and [await them](https://github.com/MetaMask/core/blob/a0687fe4e03ef1e47a13604757297760cb0dd117/packages/assets-controllers/src/TokenRatesController.ts#L390) from `detectNfts` instead of no-opping

This way the client can just `await detectNfts` and it always returns when the fetch finishes, regardless of whether it triggered a new one or hooked into an existing one.

Reply:
I like the way its done in the TokenRatesController, this would remove the updates on the NftConroller! :D